--- a/src/main.ts	2026-04-10 17:38:29
+++ b/src/main.ts	2026-04-10 17:38:55
@@ -251,8 +251,8 @@
     }
 
     const [stdout, stderr] = await Promise.all([
-      this._streamOut ? readStream(this._streamOut) : '',
-      this._streamErr ? readStream(this._streamErr) : ''
+      this._stdoutPromise ?? '',
+      this._stderrPromise ?? ''
     ]);
 
     await this._processClosed;
@@ -295,6 +295,8 @@
 
   protected _streamOut?: Readable;
   protected _streamErr?: Readable;
+  protected _stdoutPromise?: Promise<string>;
+  protected _stderrPromise?: Promise<string>;
 
   public spawn(): void {
     const cwd = getCwd();
@@ -338,9 +340,11 @@
 
     if (handle.stderr) {
       this._streamErr = handle.stderr;
+      this._stderrPromise = readStream(handle.stderr);
     }
     if (handle.stdout) {
       this._streamOut = handle.stdout;
+      this._stdoutPromise = readStream(handle.stdout);
     }
 
     this._process = handle;
